src/core/server/saved_objects/service/lib/collect_multi_namespace_references.ts

```diff
@@ -150,13 +163,22 @@ export async function collectMultiNamespaceReferences(
   const aliasesMap = await findLegacyUrlAliases(
     createPointInTimeFinder,
     objectsToFindAliasesFor,
-    ALIAS_SEARCH_PER_PAGE
+    ALIAS_OR_SHARED_ORIGIN_SEARCH_PER_PAGE
+  );
+  const objectOriginsToSearchFor = objectsWithContext
+    .filter(({ spaces }) => spaces.length !== 0)
+    .map(({ type, id, originId }) => ({ type, origin: originId || id }));
+  const originsMap = await findSharedOriginObjects(
+    createPointInTimeFinder,
+    objectOriginsToSearchFor,
+    ALIAS_OR_SHARED_ORIGIN_SEARCH_PER_PAGE
   );
```

Comment:
Yet another additional ES query we'll be performing... Not that I have an idea on how to avoid that.

Reply:
Yeah... luckily this is not an API that we expect to be called very frequently, only when users open the "Assign to spaces" flyout